tests/bluetooth/bsim_bt/bsim_test_advx/src/main.c

```diff
@@ -322,6 +323,88 @@ static void test_advx_main(void)
 
 	k_sleep(K_MSEC(1000));
 
+	printk("Re-enable advertising using extended commands (max_events)...");
+	is_sent = false;
+	num_sent_actual = 0;
+	num_sent_expected = 3;
+	ext_adv_param.timeout = 0;
+	ext_adv_param.num_events = 3;
+	err = bt_le_ext_adv_start(adv, &ext_adv_param);
+	if (err) {
+		goto exit;
+	}
+	printk("success.\n");
+
+	k_sleep(K_MSEC(100));
+
+	printk("Setting advertising random address before re-enabling...");
+	handle = 0x0000;
+	err = ll_adv_aux_random_addr_set(handle, own_addr_reenable);
+	if (err) {
+		goto exit;
+	}
+	printk("success.\n");
+
+	printk("Re-enabling...");
+	handle = 0x0000;
+	err = ll_adv_enable(handle, 1,
+			    ext_adv_param.timeout,
+			    ext_adv_param.num_events);
+	if (err) {
+		goto exit;
+	}
+	printk("success.\n");
+
+	printk("Waiting...");
+	while (!is_sent) {
+		k_sleep(K_MSEC(100));
+	}
+	printk("done.\n");
+
+	if (num_sent_actual != num_sent_expected) {
+		FAIL("Num sent actual = %u, expected = %u\n", num_sent_actual,
+		     num_sent_expected);
+	}
+
+	k_sleep(K_MSEC(1000));
+
+	printk("Re-enable advertising using extended commands (duration)...");
+	is_sent = false;
+	num_sent_actual = 0;
+	num_sent_expected = 4;
+	ext_adv_param.timeout = 50;
```

Comment:
I guess the number of sent expected is a function of the advertising interval and the timeout? 
I think this will be more clear if it was explicitly calculated, instead of implicitly:

 ```suggestion
	num_sent_expected = 4;
	ext_adv_param.timeout = 4 * ADV_INTERVAL;
```
or 
 ```suggestion
	ext_adv_param.timeout = 50;
	num_sent_expected = ext_adv_param.timeout / ADV_INTERVAL;
```

Reply:
This is to be regarded as test vector input and it cannot be calculated due to fact that advertisement intervals are random and some timeout values (if ADV_INTERVAL changed by a developer/tester) influences the number of events.

Reply:
If the interval is random and influences the number of events, how we can expect 4 advertisements to be sent?

Reply:
4 events will be transmitted within 400 to 440 ms (4 * (BT_GAP_ADV_FAST_INT_MIN_1 + random_delay)) , the limit is set at 500 ms to ensure no more than 4 event is sent. Say events are changed to 11, then 11 events transmit in 1100 to 1210 ms, calculating using interval will give a limit of 1310 ms, which means there can be 11 or 12 events occurring based on random_delay applied on each event on air.

Reply:
I see, so if the number of events increases then we might enter the possibility of have 1 more event than we expected, but that can be guarded with an __ASSERT of something to avoid that. 

But since you have already done the calculation on paper, why not bring it to the code, so that it becomes clear why we are waiting X ms for Y events? 

On another note: You mention that `4 * (BT_GAP_ADV_FAST_INT_MIN_1 + random_delay))` is 400-440ms. This implies that the random delay is 10ms, right? But `BT_GAP_ADV_FAST_INT_MIN_1` is 30 ms, so the above calculation doesn't make sense to me. Please elaborate.

Reply:
It is a bad idea to make tests and its input generic, and then add all sort of complexities. Tests being precise, deterministic and fail on any change is desired.

It was a typo, `BT_LE_ADV_CONN_NAME` uses `BT_GAP_ADV_FAST_INT_MIN_2`, 100 ms advertising interval.

Reply:
I disagree; I don't think making it harder to update or change a test should be hard. 

I think making this change will not only make it possible to update the test at a later point if wanted, but also show the calculations that are currently implicitly made (and undocumented) with regards the number of events, the timeout values and the sleep values.

Reply:
I disagree too.

> update the test at a later point if wanted

That is recipe for disaster, and only be done if software requirements mandate so. I am aware that we have not documented and approved a test procedure here, as this is a test added (not updated any existing test case) as part of development to cover the implementation changes in this PR, which is the right thing to do in software process.

That said, you may send a PR with your proposed test implementation.

If there is no further change requests and do not agree to approve, you may dismiss your review in this PR.

Reply:
Added comments.

Reply:
If I may.... I agree with @Thalley. 
Tests cases should be self explanatory. Magic numbers does not help at all. Also making tests hard to update/change is not a good way. Someone has to maintain it or change in future. Why not save effort and time to that person? 

If you verify some border conditions then you should  state it in e.g. test case name and then you can go with fixed values. Nevertheless that does not release you from requirement of documenting them (in comments or better by use of named macros). 

> That is recipe for disaster, and only be done if software requirements mandate so. 
Yes, tests are changed when the requirements are changed but why make it harder to do than it could be?

In my humble opinion there is more to do about this new block of code added to this looooooong function.

Reply:
The comments helps. I still stand by my original comment, but I not strongly enough that I will block a feature (or rather a fix to a compliance issue) because a test case may be improved for readability and refactorability.

Reply:
FWIW I agree with Piotr and Emil